eZ/Bundle/EzPublishRestBundle/Tests/Functional/ContentQueryTest.php

```diff
@@ -0,0 +1,380 @@
+    </LogicalOr>
+  </Query>  
+  <limit>10</limit>  
+  <offset>0</offset> 
+</ContentQuery>
+</ViewInput>
+XML;
+
+        $request->setContent($body);
+
+        $response = $this->sendHttpRequest($request);
+        self::assertHttpResponseCodeEquals($response, 200);
+        $jsonResponse = json_decode($response->getContent());
+        self::assertEquals(2, $jsonResponse->View->Result->count);
+    }
+
+    private function createTestContentType(): string
+    {
+        $body = <<< XML
+<?xml version="1.0" encoding="UTF-8"?>
+<ContentTypeCreate>
+  <identifier>tags-test</identifier>
+  <names>
+    <value languageCode="eng-GB">testContentQueryWithTags</value>
+  </names>
+  <remoteId>testContentQueryWithTags</remoteId>
+  <urlAliasSchema>&lt;title&gt;</urlAliasSchema>
+  <nameSchema>&lt;title&gt;</nameSchema>
+  <isContainer>true</isContainer>
+  <mainLanguageCode>eng-GB</mainLanguageCode>
+  <defaultAlwaysAvailable>true</defaultAlwaysAvailable>
+  <defaultSortField>PATH</defaultSortField>
+  <defaultSortOrder>ASC</defaultSortOrder>
+  <FieldDefinitions>
+    <FieldDefinition>
+      <identifier>title</identifier>
+      <fieldType>ezstring</fieldType>
+      <fieldGroup>content</fieldGroup>
+      <position>1</position>
+      <isTranslatable>true</isTranslatable>
+      <isRequired>true</isRequired>
+      <isInfoCollector>false</isInfoCollector>
+      <defaultValue>New Title</defaultValue>
+      <isSearchable>true</isSearchable>
+      <names>
+        <value languageCode="eng-GB">Title</value>
+      </names>
+      <descriptions>
+        <value languageCode="eng-GB">This is the title</value>
+      </descriptions>
+    </FieldDefinition>
+    <FieldDefinition>
+      <identifier>tags</identifier>
+      <fieldType>ezkeyword</fieldType>
+      <fieldGroup>content</fieldGroup>
+      <position>2</position>
+      <isTranslatable>true</isTranslatable>
+      <isRequired>true</isRequired>
+      <isInfoCollector>false</isInfoCollector>
+      <isSearchable>true</isSearchable>
+      <names>
+        <value languageCode="eng-GB">Tags</value>
+      </names>
+      <descriptions>
+        <value languageCode="eng-GB">Those are searchable tags</value>
+      </descriptions>
+    </FieldDefinition>
+   </FieldDefinitions>
+</ContentTypeCreate>
+XML;
+
+        $request = $this->createHttpRequest(
+            'POST',
+            '/api/ezp/v2/content/typegroups/1/types?publish=true',
+            'ContentTypeCreate+xml',
+            'ContentType+json'
+        );
+        $request->setContent($body);
+        $response = $this->sendHttpRequest($request);
+
+        return $response->getHeader('Location');
+    }
+
+    private function createTestContentWithTags(string $name, array $tags): string
+    {
+        $request = $this->createHttpRequest('POST', '/api/ezp/v2/content/objects', 'ContentCreate+xml', 'ContentInfo+json');
+        $tagsString = implode(',', $tags);
+        $body = <<< XML
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
You have a lot of these XML's. Could you consider generating them from reusable methods? Not necessarily single one that does everything, but it seems to me that at least some parts of XML could be generated by common method.

Also in case of tests doing the same thing, like setting input, sending request and comparing input with expected one, maybe it is worth to explore data providers instead of multiple tests? I'm not sure however if this is applicable here, just saying... :)